File: app/src/main/java/net/vornao/ddns/dome/fragments/BottomSheetFragment.java
package net.vornao.ddns.dome.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import net.vornao.ddns.dome.R;
import net.vornao.ddns.dome.shared.Const;

public class BottomSheetFragment extends BottomSheetDialogFragment {

    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.device_details_bottomsheet, container, false);

    }

    @Override public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // TextView deviceIP = view.findViewById(R.id.ddevice_ip);
        TextView deviceName = view.findViewById(R.id.ddevice_name);
        TextView devicePubTopic = view.findViewById(R.id.ddevice_pub_topic);
        TextView deviceSubTopic = view.findViewById(R.id.ddevice_sub_topic);

        // deviceIP.setText(String.format(getString(R.string.ip_address_info), "192.168.173.34"));
        try {
            assert getArguments() != null;
        }catch (AssertionError e){
            return;
        }
        deviceName.setText(String.format(getString(R.string.device_id_info), getArguments().getString(Const.DEVICE_ID)));
        deviceSubTopic.setText(String.format(getString(R.string.device_sub_info), getArguments().getString(Const.MQTT_TOPIC_C),getArguments().getString(Const.DEVICE_ID)));
        devicePubTopic.setText(String.format(getString(R.string.device_pub_info), getArguments().getString(Const.MQTT_TOPIC_C)));
    }

}

File: app/src/main/java/net/vornao/ddns/dome/fragments/BottomSheetFragment.kt
package net.vornao.ddns.dome.fragments

import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import net.vornao.ddns.dome.R
import android.widget.TextView
import com.google.android.material.snackbar.Snackbar
import android.widget.LinearLayout
import android.content.SharedPreferences
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.EditTextPreference.OnBindEditTextListener
import android.widget.EditText
import android.text.InputType
import android.content.Intent
import android.view.View
import net.vornao.ddns.dome.activities.AboutActivity
import net.vornao.ddns.dome.shared.Const
import java.lang.AssertionError

class BottomSheetFragment : BottomSheetDialogFragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.device_details_bottomsheet, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // TextView deviceIP = view.findViewById(R.id.ddevice_ip);
        val deviceName = view.findViewById<TextView>(R.id.ddevice_name)
        val devicePubTopic = view.findViewById<TextView>(R.id.ddevice_pub_topic)
        val deviceSubTopic = view.findViewById<TextView>(R.id.ddevice_sub_topic)

        // deviceIP.setText(String.format(getString(R.string.ip_address_info), "192.168.173.34"));
        try {
            assert(arguments != null)
        } catch (e: AssertionError) {
            return
        }
        deviceName.text = String.format(getString(R.string.device_id_info), requireArguments().getString(Const.DEVICE_ID))
        deviceSubTopic.text = String.format(getString(R.string.device_sub_info), requireArguments().getString(Const.MQTT_TOPIC_C), requireArguments().getString(Const.DEVICE_ID))
        devicePubTopic.text = String.format(getString(R.string.device_pub_info), requireArguments().getString(Const.MQTT_TOPIC_C))
    }
}